otx/algorithms/classification/tasks/openvino.py

```diff
@@ -221,25 +225,81 @@ def infer(
+
+            probs_meta = TensorEntity(name="probabilities", numpy=probs.reshape(-1))
+            dataset_item.append_metadata_item(probs_meta, model=self.model)
+
             feature_vec_media = TensorEntity(name="representation_vector", numpy=repr_vector.reshape(-1))
             dataset_item.append_metadata_item(feature_vec_media, model=self.model)
             if dump_features:
-                saliency_media = ResultMediaEntity(
-                    name="Saliency Map",
-                    type="saliency_map",
-                    annotation_scene=dataset_item.annotation_scene,
-                    numpy=actmap,
-                    roi=dataset_item.roi,
-                )
-                dataset_item.append_metadata_item(saliency_media, model=self.model)
+                if saliency_map.ndim == 2:
+                    # Single saliency map per image, support e.g. EigenCAM use case
+                    actmap = get_actmap(saliency_map, (dataset_item.width, dataset_item.height))
+                    saliency_media = ResultMediaEntity(
+                        name="Saliency Map",
+                        type="saliency_map",
+                        annotation_scene=dataset_item.annotation_scene,
+                        numpy=actmap,
+                        roi=dataset_item.roi,
+                    )
+                    dataset_item.append_metadata_item(saliency_media, model=self.model)
+                elif saliency_map.ndim == 3:
+                    # Multiple saliency maps per image (class-wise saliency map), support e.g. Recipro-CAM use case
+                    predicted_labels = set()
+                    for scored_label in predicted_scene.annotations[0].get_labels():
+                        predicted_labels.add(scored_label.label)
+
+                    for class_id, class_wise_saliency_map in enumerate(saliency_map):
+                        label = self.task_environment.get_labels()[class_id]
+                        if label in predicted_labels:
+                            # TODO (negvet): Support more advanced use case,
+                            #  when all/configurable set of saliency maps is returned
+                            actmap = get_actmap(class_wise_saliency_map, (dataset_item.width, dataset_item.height))
+                            saliency_media = ResultMediaEntity(
+                                name=label.name,
+                                type="saliency_map",
+                                annotation_scene=dataset_item.annotation_scene,
+                                numpy=actmap,
+                                roi=dataset_item.roi,
+                            )
+                            dataset_item.append_metadata_item(saliency_media, model=self.model)
+                else:
+                    raise RuntimeError(
+                        f"Single saliency map has to be 2 or 3-dimensional, " f"but got {saliency_map.ndim} dims"
+                    )
```

Comment:
Can we replace this block by add_saliency_maps_to_dataset_item(), too?

Reply:
Yes I'll integrate it too